scripts/extract_business_rules.py

```
@@ -0,0 +1,201 @@
+            self.generic_visit(node)
+
+    Visitor().visit(tree)
+    return rules
+
+def find_test_file(domain_filepath):
+    parts = domain_filepath.split(os.sep)
+    if 'domains' in parts:
+        idx = parts.index('domains')
+        test_parts = parts[:idx] + ['tests', 'unit', 'domain'] + parts[idx+1:]
+        test_parts[-1] = 'test_' + test_parts[-1]
+        
```

Pylint (4.0.5)

[convention] Trailing whitespace

(C0303)